lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +218,110 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private nexusEndPointRepoFrom(url: string): {
+    nexusRegistry?: string;
+    repository?: string;
+  } {
+    const endPointRe = regEx(
+      `^(?<nexusRegistry>.*:\\d{1,5}).*\\/repository\\/(?<repository>\\w+)\\/?`
+    );
+    const { nexusRegistry, repository } = url.match(endPointRe)?.groups ?? {};
+    return { nexusRegistry, repository };
+  }
+
+  @cache({
+    namespace: 'datasource-nexus',
+    key: (nexusEndPoint: string) => `isNexusDataSource:${nexusEndPoint}`,
+    ttlMinutes: 1440,
+  })
+  async isNexusDataSource(nexusEndPoint: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const nexusUrl = joinUrlParts(nexusEndPoint, statusEndPoint);
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      return this.extractServerFrom(response).startsWith('Nexus/') ?? false;
+    } catch (err) {
+      // istanbul ignore if
```

Comment:
why no test?

Reply:
The `HttpHeaders` interface is saying that server could be a string or string array.
I managed to mock a string response but when I try to mock array response for code converge it converts the array to a string :
```
      httpMock
        .scope('http://localhost:8081/')
        .get('/service/rest/v1/status')
        .reply(200, {}, { server: ['s1', 's2', 's3'] })
```
Oddly the response from this mock is `server : 's1, s2, s3'` instead of `server : ['s1, 's2', 's3']`. Have you come across this before ?